servatrice/src/servatrice_database_interface.cpp

```diff
@@ -1091,3 +1091,23 @@ QList<ServerInfo_ChatMessage> Servatrice_DatabaseInterface::getMessageLogHistory
 
     return results;
 }
+
+bool Servatrice_DatabaseInterface::isUserADonator(const QString &user)
+{
+    if (!checkSql())
+        return false;
+    
+    QSqlQuery *query = prepareQuery("SELECT donator from {prefix}_users WHERE name = :user_name");
```

Comment:
The query is to identify if the user is flagged as a "donator" (the naming could be changed).  The query is to be used to take some sort of action on the information.  It just so happens in this change it is for granting access to the user in the event the server is full.   Nothing is to say the information couldnt be used else were for other reasons.

The current way the login works:
connection is made -> db queried for capacity -> db is queried for password correct -> return response.

This change make it looks something like this:
connection is made -> session is created -> db is queried for user password correct -> db is queried for donator status -> db is queried for capacity -> return response. 

If you want I could probably combine the "db queried for password correctness & db is queried for donator status" and create a new AuthenticationResults and then action against that.  It would probably reduce the amount of queries by 1 (haven't looked so not sure).  Ill take a look.